src/query/builder.js

```diff
@@ -1061,6 +1061,42 @@ assign(Builder.prototype, {
     return this;
   },
 
+  // Skips locked rows when using a lock constraint.
+  skipLocked() {
+    const { _method } = this;
+    if (!includes(['pluck', 'first', 'select'], _method)) {
+      throw new Error(`Cannot chain .skipLocked() on "${_method}" query!`);
+    }
+    if (!includes(['forShare', 'forUpdate'], this._single.lock)) {
+      throw new Error(
+        '.skipLocked() can only be used after a call to .forShare() or .forUpdate()!'
+      );
+    }
+    if (this._single.waitMode) {
+      throw new Error('.skipLocked() cannot be used with .noWait()!');
+    }
+    this._single.waitMode = 'skipLocked';
+    return this;
+  },
+
+  // Causes error when acessing a locked row instead of waiting for it to be released.
+  noWait() {
+    const { _method } = this;
+    if (!includes(['pluck', 'first', 'select'], _method)) {
```

Comment:
any reasons why duplicate code can't be reused? do you envision it branching out eventually?

Reply:
I'm actually not sure how to best handle this. I just copied this piece from the first() method, since all databases only allow "skip locked" and "nowait" on select queries. I could, however, just leave it up for the database driver to return the error directly to the calling code, without validating it first.

Reply:
@kibertoad handled in commit 18cccab, what do you think?

Reply:
I'm also thinking about moving the "isSelectQuery" checks to the proper "forShare" and "forUpdate" methods